deep_quoridor/src/agents/alphazero/nn_evaluator.py

```diff
@@ -36,20 +38,20 @@ def evaluate(self, game: Quoridor):
 
         self.network.eval()  # Disables dropout
 
+        # Prepare a tensor with True only on INVALID action positions so that we can set those values to
+        # a large negative number below, to discourage choosing them.
```

Comment:
I don't think that this comment quite describes what the following lines do. They're creating a tensor with the valid indices as integers, eg. instead of `[True, False, True, True]` it would contain `[0, 2, 3]` if i understand correctly. This could explain why you had trouble getting ignore_indices to do what you expected.